Local War/Assets/Resources/Script/Weapon/Weapon.cs

```diff
@@ -1,20 +1,55 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.Networking;
 
 public class Weapon : MonoBehaviour {
-    protected int       id;
-    protected int       type;
-    protected string    description;
-    protected Camera    playerCame;
+    protected int           id;
+    protected int           type;
+    protected string        description;
+    protected WeaponSys     playerWeaponSys;
+    protected bool          isShooting;
+    protected bool          isFinishShoot;
```

Comment:
Why is shoot in weapon. Weapon should not have a concept of shooting. It could be either a ranged weapon or a melee weapon. All of this code could and probably should go into ranged weapon.